### PR TITLE
Rename module and package from MemSQL to SingleStore

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Rename module and package from MemSQL to SingleStore

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

### CODE REVIEW

**Review by hashhar — COMMENTED** [review]
There are some remaining usages:

MEMSQL_* constants in SingleStoreClient.
Some configs in SingleStoreConfig.

Rest of them are in test classes like comments and the `DataSetup` implementations.

File: plugin/trino-singlestore/src/test/java/io/trino/plugin/singlestore/SingleStoreQueryRunner.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public static DistributedQueryRunner createMemSqlQueryRunner(
 
             server.execute("CREATE SCHEMA tpch");
 
-            queryRunner.installPlugin(new MemSqlPlugin());
+            queryRunner.installPlugin(new SingleStorePlugin());
             queryRunner.createCatalog("memsql", "memsql", connectorProperties);
```
Comment (hashhar) [inline plugin/trino-singlestore/src/test/java/io/trino/plugin/singlestore/SingleStoreQueryRunner.java:58]: Let's use new name here.
Old name is already covered by a test in TestSingleStorePlugin.

File: plugin/trino-singlestore/src/main/java/io/trino/plugin/singlestore/SingleStoreConfig.java [hunk 2]
```diff
@@ -32,7 +32,7 @@ public boolean isAutoReconnect()
     }
 
     @Config("memsql.auto-reconnect")
```
Comment (hashhar) [inline plugin/trino-singlestore/src/main/java/io/trino/plugin/singlestore/SingleStoreConfig.java:34]: legacy config these and add new ones with singlestore prefix?

**Comment by ebyhr** [comment]
@hashhar I didn't rename them intentionally to get clean diff.

**Review by hashhar — APPROVED** [review]
LGTM. We can follow-up on non-user facing stuff later.

**Comment by hashhar** [comment]
I pushed a fix for failed tests, verified everything passes locally. Change is https://github.com/trinodb/trino/pull/11482/files#diff-c8a7eb785c3399d1d0c3a88518918a10c40911ed811c7d2328d9bdffe5a12b62R452-R462

**Pull request merged**